### PR TITLE
docs: Improve nrfcloud logging docs

### PR DESCRIPTION
**Pull request by plskeggs**
Mention log and alert use within the nrfcloud_mqtt_multi_service sample.

Document binary logs.

Change default for MMSS to use JSON format.

### CODE REVIEW

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 1]
```diff
@@ -8,7 +8,7 @@ nRF9160: nRF Cloud MQTT multi-service
    :depth: 2
 
 This sample is a minimal, error tolerant, integrated demonstration of the :ref:`lib_nrf_cloud`, :ref:`lib_location`, and :ref:`lib_at_host` libraries.
-It demonstrates how you can use these libraries together to support Firmware-Over-The-Air (FOTA), Location Services, periodic sensor samples, and more in your `nRF Cloud`_-enabled application.
+It demonstrates how you can use these libraries together to support Firmware-Over-The-Air (FOTA), Location Services, Alert and Log Services, transmit periodic sensor samples, and more in your `nRF Cloud`_-enabled application.
```
Comment (glarsennordic) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:11]: nit: `transmission of periodic sensor samples` or `periodic sensor sampling`

Comment (glarsennordic) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:11]: However, the sentence as a whole could probably be made less awkward:

> It demonstrates how you can integrate Firmware-Over-The-Air (FOTA), Location Services, Alert and Log Services, periodic sensor sampling, and more into your `nRF Cloud`_-enabled application.

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 2]
```diff
@@ -40,6 +40,9 @@ This sample implements or demonstrates the following features:
 * Transmission of sensor and GNSS location samples to the nRF Cloud portal as `nRF Cloud device messages <nRF Cloud Device Messages_>`_.
 * Construction of valid `nRF Cloud device messages <nRF Cloud Device Messages_>`_ using `cJSON`_.
 * Minimal LED status indication using the `Zephyr LED API`_.
+* Sends an alert using the :ref:`lib_nrf_cloud_alert` library when the sample starts and again whenever the temperature limit is exceeded.
+* Optionally sends logging messages to the cloud using the :ref:`lib_nrf_cloud_log` library.
```
Comment (glarsennordic) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:44]:
```suggestion
* Transmission of an alert on sample startup, using the :ref:`lib_nrf_cloud_alert` library.
* Transmission of additional alerts, whenever a specified temperature limit is exceeded.
* Optional transmission of log messages to the cloud using the :ref:`lib_nrf_cloud_log` library.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 3]
```diff
@@ -598,6 +601,22 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
 
 See also :ref:`the paragraphs on the Wi-Fi location tracking method <nrf_cloud_mqtt_multi_service_wifi_location_tracking>`.
 
+Building with nRF Cloud logging support
+=======================================
+
+To enable `Zephyr Logging`_ to nRF Cloud using the :ref:`lib_nrf_cloud_log` library, add the following parameter to your build command:
```
Comment (glarsennordic) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:607]:
```suggestion
To enable transmission of `logs <Zephyr Logging_>`_ to nRF Cloud using the :ref:`lib_nrf_cloud_log` library, add the following parameter to your build command:
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 4]
```diff
@@ -598,6 +601,22 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
...
+
+``-DOVERLAY_CONFIG=overlay_nrfcloud_logging.conf``
+
+This overlay allows the sample and various subsystems that have logging enabled to send their logs to nRF Cloud.
```
Comment (glarsennordic) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:611]:
```suggestion
This overlay enables transmission of `logs <Zephyr Logging_>`_ to nRF Cloud.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 5]
```diff
@@ -598,6 +601,22 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
...
+Set the :kconfig:option:`CONFIG_NRF_CLOUD_LOG_OUTPUT_LEVEL` option to the log level of messages to send to nRF Cloud, such as ``4`` for debug log messages.
+Set the :kconfig:option:`CONFIG_NRF_CLOUD_REST_DEVICE_MESSAGE_SAMPLE_LOG_LEVEL_DBG` option so that log messages are generated on each button press.
```
Comment (glarsennordic) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:613]: I think this line was included by mistake

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 6]
```diff
@@ -598,6 +601,22 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
...
+
+By default, the overlay sends log messages to the cloud in JSON format.
```
Comment (glarsennordic) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:615]: Is this true?

**Review by glarsennordic — CHANGES_REQUESTED** [review]

File: doc/nrf/libraries/networking/nrf_cloud_log.rst [hunk 7]
```diff
@@ -101,6 +101,12 @@ Configure the message encoding:
 * :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` or :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY`
 
 See `Dictionary-based Logging`_ to learn how dictionary-based logging works, how the dictionary is built, and how to decode the binary log output.
+Dictionary logs are compact binary log messages which require decoding using an offline script.
+As such, dictionary logs are up to 3 times smaller than JSON logs, but cannot be viewed in the nRF Cloud user interface in real time.
+Instead, the user interface displays a link from which you can download a single .bin file containing binary logs.
+Note that to successfully decode dictionary logs, you must use the :file:`log_dictionary.json` file built by the build system at the same time as the firmware image.
+If you modify the source code and build the firmware image again, the :file:`log_dictionary.json` file may change.
+You must be careful to keep track of each firmare image and :file:`log_dictionary.json` file, as well as when a device runs each different firmware image.
```
Comment (jayteemo) [inline doc/nrf/libraries/networking/nrf_cloud_log.rst:109]:
```suggestion
You must be careful to keep track of each firmware image and :file:`log_dictionary.json` file, as well as when a device runs each different firmware image.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst (hunk 1 above)
Comment (plskeggs) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:11]: Awkward?  Little o' me?  Ha ha.

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst (hunk 6 above)
Comment (plskeggs) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:616]: I changed the nrfcloud log overlay in this PR from binary to JSON mode.

But the overlay is not used by default.  Is that what you thought I meant?

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf [hunk 8]
```diff
@@ -18,9 +18,8 @@ CONFIG_LOG_BUFFER_SIZE=4096
 # Select which format to generate. Dictionary logs are compact but require
 # decoding on the server or user PC.  Text logs are verbose but immediately
 # visible in the nRF Cloud web portal.
-CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY=y
-#CONFIG_NRF_CLOUD_LOG_DICT_JSON=y
-#CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT=y
+#CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY=y
```
Comment (jayteemo) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf:21]: while you are in these files:
https://github.com/nrfconnect/sdk-nrf/commit/9af0e3d2dc0673e7638621b9b2ea020e1c4eafc6#r116082219

Comment (plskeggs) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf:21]: Thanks, I missed your comment.

**Review by glarsennordic — APPROVED** [review]
LGTM!

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf (hunk 8 above)
Comment (jayteemo) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf:21]:
```
CONFIG_NRF_CLOUD_LOG_LEVEL_DBG=y
CONFIG_NRF_CLOUD_LOG_LOG_LEVEL_INF=y
```
is that what you want for the overlay?

**Review by jayteemo — APPROVED** [review]

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf (hunk 8 above)
Comment (plskeggs) [reply samples/nrf9160/nrf_cloud_mqtt_multi_service/overlay_nrfcloud_logging.conf:21]: This one I'll remove: CONFIG_NRF_CLOUD_LOG_LEVEL_DBG=y

This one sets the log level for the cloud logging system: CONFIG_NRF_CLOUD_LOG_LOG_LEVEL_INF=y.  I'd like to keep it that way.

File: doc/nrf/libraries/networking/nrf_cloud_log.rst [hunk 9]
```diff
@@ -101,6 +101,12 @@ Configure the message encoding:
 * :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` or :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY`
 
 See `Dictionary-based Logging`_ to learn how dictionary-based logging works, how the dictionary is built, and how to decode the binary log output.
+Dictionary logs are compact binary log messages which require decoding using an offline script.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_log.rst:104]:
```suggestion
Dictionary logs are compact binary log messages that require decoding using an offline script.
```

File: doc/nrf/libraries/networking/nrf_cloud_log.rst [hunk 10]
```diff
@@ -101,6 +101,12 @@ Configure the message encoding:
...
+As such, dictionary logs are up to 60% smaller than JSON logs, but cannot be viewed in the nRF Cloud user interface in real time.
+Instead, the user interface displays a link from which you can download a single .bin file containing binary logs.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_log.rst:106]:
```suggestion
Instead, the user interface displays a link from which you can download a single binary file containing the logs.
```

File: doc/nrf/libraries/networking/nrf_cloud_log.rst [hunk 11]
```diff
@@ -101,6 +101,12 @@ Configure the message encoding:
...
+Note that to successfully decode dictionary logs, you must use the :file:`log_dictionary.json` file built by the build system at the same time as the firmware image.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_log.rst:107]:
```suggestion
To successfully decode dictionary logs, you must use the :file:`log_dictionary.json` file built by the build system at the same time as the firmware image.
```

File: doc/nrf/libraries/networking/nrf_cloud_log.rst [hunk 12]
```diff
@@ -101,6 +101,12 @@ Configure the message encoding:
...
+If you modify the source code and build the firmware image again, the :file:`log_dictionary.json` file may change.
+You must be careful to keep track of each firmware image and :file:`log_dictionary.json` file, as well as when a device runs each different firmware image.
```
Comment (peknis) [inline doc/nrf/libraries/networking/nrf_cloud_log.rst:109]:
```suggestion
Keep track of each firmware image and the :file:`log_dictionary.json` file when a device runs different firmware images.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 13]
```diff
@@ -172,7 +172,10 @@ Bluetooth mesh samples
 nRF9160 samples
 ---------------
 
-|no_changes_yet_note|
+* :ref:`nrf_cloud_mqtt_multi_service` sample:
+
+  * Added documentation for use of the :ref:`lib_nrf_cloud_alert` and :ref:`lib_nrf_cloud_log` libraries.
```
Comment (peknis) [inline doc/nrf/releases/release-notes-changelog.rst:177]:
```suggestion
  * Added documentation for using the :ref:`lib_nrf_cloud_alert` and :ref:`lib_nrf_cloud_log` libraries.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 14]
```diff
@@ -40,6 +40,10 @@ This sample implements or demonstrates the following features:
 * Transmission of sensor and GNSS location samples to the nRF Cloud portal as `nRF Cloud device messages <nRF Cloud Device Messages_>`_.
 * Construction of valid `nRF Cloud device messages <nRF Cloud Device Messages_>`_ using `cJSON`_.
 * Minimal LED status indication using the `Zephyr LED API`_.
+* Transmission of an alert on sample startup, using the :ref:`lib_nrf_cloud_alert` library.
```
Comment (peknis) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:43]:
```suggestion
* Transmission of an alert on sample startup using the :ref:`lib_nrf_cloud_alert` library.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 15]
```diff
@@ -598,6 +602,21 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
...
 
+Building with nRF Cloud logging support
+=======================================
+
+To enable transmission of `logs <Zephyr Logging_>`_ to nRF Cloud using the :ref:`lib_nrf_cloud_log` library, add the following parameter to your build command:
+
+``-DOVERLAY_CONFIG=overlay_nrfcloud_logging.conf``
+
+This overlay enables transmission of `logs <Zephyr Logging_>`_ to nRF Cloud.
+Set the :kconfig:option:`CONFIG_NRF_CLOUD_LOG_OUTPUT_LEVEL` option to the log level of messages to send to nRF Cloud, such as ``4`` for debug log messages.
```
Comment (peknis) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:613]:
```suggestion
Set the :kconfig:option:`CONFIG_NRF_CLOUD_LOG_OUTPUT_LEVEL` Kconfig option to the log level of messages to send to nRF Cloud, such as ``4`` for debug log messages.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 16]
```diff
@@ -598,6 +602,21 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
...
+
+The overlay selects the :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` option which enables log messages in JSON format.
```
Comment (peknis) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:615]:
```suggestion
The overlay selects the :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` Kconfig option that enables log messages in JSON format.
```

File: samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst [hunk 17]
```diff
@@ -598,6 +602,21 @@ This is only supported on the `Nordic nRF9160 DK`_ with an attached nRF7002 EK.
...
+You can read JSON log messages in real-time in the nRF Cloud user interface.
+However, because JSON logs are large, you may want to edit the overlay file to change to using dictionary logging.
+To do so, deselect the :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` option and select the :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY` option.
```
Comment (peknis) [inline samples/nrf9160/nrf_cloud_mqtt_multi_service/README.rst:618]:
```suggestion
Deselect the :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_TEXT` Kconfig option and select :kconfig:option:`CONFIG_LOG_BACKEND_NRF_CLOUD_OUTPUT_DICTIONARY` instead.
```

**Review by peknis — CHANGES_REQUESTED** [review]

**Comment by plskeggs** [comment]
@peknis made your changes, thanks.

Please re-review/approve.

**Review by peknis — APPROVED** [review]

**Pull request merged**